src/components/messageArchive/messageArchive.tsx

```diff
@@ -0,0 +1,236 @@
+import './messageArchive.css'
+
+import Alert from '@mui/material/Alert'
+import Chip from '@mui/material/Chip'
+import Box from '@mui/system/Box'
+import React, { useEffect, useRef, useState } from 'react'
+
+import ElementRenderer from '../elementRenderer/elementRenderer'
+import Icon from '../icon/icon'
+import MessageCanvas, {
+  type MessageContainerProps,
+} from '../messageCanvas/messageCanvas'
+import type { ComponentMap, Message } from '../types'
+
+export interface MessageArchiveProps extends MessageContainerProps {
+  /** A component map contains message formats as keys and their corresponding React components as values. */
+  supportedElements: ComponentMap
+  /** Previous chat messages */
+  receivedMessages: Message[]
```

Comment:
Both Dev and end user experience would be better if expected a method `fetchOldMessages:() =>Promise<Message[]>` and an optional loadingIcon. It will make it easier for developers to define and provide custom fetch logic. The component could show the loading icon until the response is received.